connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -136,10 +148,51 @@ object functions {
       case v: Array[Byte] => createLiteral(_.setBinary(ByteString.copyFrom(v)))
       case v: collection.mutable.WrappedArray[_] => lit(v.array)
       case v: LocalDate => createLiteral(_.setDate(v.toEpochDay.toInt))
+      case v: UTF8String => createLiteral(_.setString(v.toString))
+      case v: Decimal => createDecimalLiteral(Math.max(v.precision, v.scale), v.scale, v.toString)
+      case v: Instant => createLiteral(_.setTimestamp(DateTimeUtils.instantToMicros(v)))
+      case v: Timestamp => createLiteral(_.setTimestamp(DateTimeUtils.fromJavaTimestamp(v)))
+      case v: LocalDateTime =>
+        createLiteral(_.setTimestampNtz(DateTimeUtils.localDateTimeToMicros(v)))
+      case v: Date => createLiteral(_.setDate(DateTimeUtils.fromJavaDate(v)))
+      case v: Duration => createLiteral(_.setDayTimeInterval(IntervalUtils.durationToMicros(v)))
+      case v: Period => createLiteral(_.setYearMonthInterval(IntervalUtils.periodToMonths(v)))
+      case v: CalendarInterval => createCalendarIntervalLiteral(v.months, v.days, v.microseconds)
       case null => createLiteral(_.setNull(nullType))
+      case v: Literal => literalToColumn(v)
       case _ => unsupported(s"literal $literal not supported (yet).")
     }
   }
+
+  private def literalToColumn(literal: Literal): Column = {
```

Comment:
We don't need this. In fact for the next Spark release we will be removing the Catalyst dependency.

Reply:
So, we should remove `case v: Literal => literalToColumn(v)` ?

Reply:
Yeah we should. Sorry about that.

Reply:
ok, let me update the code